mypy_django_plugin/transformers/managers.py

```diff
@@ -105,6 +109,8 @@ def get_funcdef_type(definition: Union[FuncBase, Decorator, None]) -> Optional[P
             ret_type = Instance(queryset_info, [manager_instance.args[0], manager_instance.args[0]])
         variables = []
 
+    ret_type = _find_type_var_from_mro(ret_type, queryset_info)
```

Comment:
I'm wondering why we need to dig through the mro again, once we know which base has the method?

I think it should suffice to look through `base_that_has_method.args` for `ret_type`? Or what am I missing?

What I mean is that if one were to write it out. The type var in a method signature would be found in the arguments of the class it's declared in.

Reply:
I checked out your branch and my suspicions were correct, your tests pass if we just call `_manage_type_var` for `base_that_has_method`

```suggestion
    ret_type = _manage_type_var(ret_type, base_that_has_method)
```

We've already dug through the mro to find everything. There was no need to do it again

Reply:
In this example :
```
class _MyModelQuerySet(models.QuerySet[_CTE]):
    def example(self) -> _CTE: ...

class _MyModelQuerySet2(_MyModelQuerySet[_CTE2]):
    def example_2(self) -> _CTE_2: ...

class MyModelQuerySet(_MyModelQuerySet2["MyModel"]):
    ...
```

base_that_has_method will be `titi._MyModelQuerySet[_CTE_2`1]`.


base_that_has_method.args is thus `(_CTE_2`1,)`.

So we do not have anywere MyModel :).

Or maybe I missed something.

Reply:
> _manage_type_var(ret_type, base_that_has_method)

It does not work for me :

![image](https://github.com/typeddjango/django-stubs/assets/26866746/b0a11bb8-8747-4a04-ac79-e72bf8b8e95b)

You launched the test `handles_subclasses_of_subclasses_of_queryset` ?

Reply:
No, it doesn't work for me either.